### PR TITLE
Output image to term

### PR DESCRIPTION
**Pull request by Ruxton**
Adds a plugin to output captured image to the terminal per https://www.iterm2.com/documentation-images.html

### CODE REVIEW

File: lib/lolcommits/plugins/term_output.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# -*- encoding : utf-8 -*-
+require 'rest_client'
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/term_output.rb]: don't think you need the rest-client here

File: lib/lolcommits/plugins/term_output.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        puts 'Repo is empty, skipping output'
+      else
+        base64 = Base64.encode64(open(runner.main_image) { |io| io.read })
+
+        puts "\e]1337;File=inline=1:#{base64};alt=#{runner.message}\a\n"
+      end
+
+    end
+
+    def configured?
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/term_output.rb]: I think you can use the default logic from plugin.rb [here](https://github.com/mroth/lolcommits/blob/master/lib/lolcommits/plugin.rb#L77)

File: lib/lolcommits/plugins/term_output.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# -*- encoding : utf-8 -*-
+require 'rest_client'
+require 'base64'
+
+module Lolcommits
+  class TermOutput < Plugin
+
+    def initialize(runner)
+      super
+      options
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/term_output.rb]: I think this call to the options hash is not needed either...

**Comment by matthutchinson** [comment]
Nice work, just a couple of inline comments above.. Also, if this will only work on iTerm2, is there anyway you can check (possibly inside `run_postcapture`) that you are running in that terminal?  E.g. the $TERM (or some other) ENV var?

File: lib/lolcommits/plugins/term_output.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    def run_postcapture
+      return unless valid_configuration?
+
+      if !runner.vcs_info || runner.vcs_info.repo.empty?
+        puts 'Repo is empty, skipping output'
+      else
+        base64 = Base64.encode64(open(runner.main_image) { |io| io.read })
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/term_output.rb]: Travis / Rubocop is complaining about style [here](https://travis-ci.org/mroth/lolcommits/jobs/147363210)

**Comment by Ruxton** [comment]
It works in some other terms too, any terminal that does not recognise it should just snuff the output.

Will work on fixes soon.

**Comment by matthutchinson** [comment]
👍 OK - I'll double check on some other terminals before merging, (including windows 😷)

**Comment by matthutchinson** [comment]
Any update on this guy? I have some time free tomorrow and could push it over the line if your like (the rubocop stuff and extra testing ...)

**Comment by Ruxton** [comment]
I think I've fixed the style warnings, haven't checked it 100% yet.

Still need to make sure it doesn't break hard on window.

**Comment by matthutchinson** [comment]
💚 now, i'll review, and try on windows - then merge if all OK

**Comment by matthutchinson** [comment]
So I had a go with Windoze (10 as a virtual machine -- setup with our [standard guide](https://github.com/mroth/lolcommits/wiki/Installing-on-Windows)

Unfortunately with this plugin enabled, I get the following dumped to the Windows cmd line..

screen shot 2016-08-23 at 22 36 31

To mitigate this, you could use a `Lolcommits::Platform.platform_windows?` check within the plugin (or even in the plugin's configure method - i.e. don't allow the plugin to be enabled=true if on Windows - aborting with a message saying unix/mac only)

Also, in the regular mac terminal, with the plugin enabled, I get one extra empty line in the output (which I guess is OK)..

screen shot 2016-08-23 at 22 58 09

**Comment by matthutchinson** [comment]
HI @Ruxton - any update on this guy? If you don't mind - I can take if from here and push it through (putting in that check for windows, and removing the newline on OSX Terminal)

**Comment by Ruxton** [comment]
@matthutchinson  that would be awesome if you could do that, pretty swamped atm

**Comment by matthutchinson** [comment]
No problem, will do today 👍

**Comment by matthutchinson** [comment]
Clo

**Comment by matthutchinson** [comment]
@Ruxton - I've just merged #323 for this, will get a release out soon. I made changes to only run this if you're in iTerm and some extra escape sequence stuff for tmux terminals.

**Comment by matthutchinson** [comment]
Hi @Ruxton 👋

So you may (or may not) have noticed that we [recently extracted](https://github.com/mroth/lolcommits/issues?utf8=✓&q=label%3Aarchitecture+label%3Aplugin+) all lolcommit plugins to external gems!

🏅 Since you contributed this term_output plugin, I've added you (as an admin) to the new [lolcommits-term_output](https://github.com/lolcommits/lolcommits-term_output) repo.

Of course **you are under no obligation** to maintain this code or gem going forward. This is just a quick message to explain what has happened and give you the opportunity to take ownership again.

If you're interested in picking this up, **send me your email address** (to matt/at/hiddeloop.com) and I'll add you as a new gem owner (with publishing rights) [here](https://rubygems.org/gems/lolcommits-term_output).

Otherwise I will continue to maintain the gem, and make sure things are kept up to date.

---

To get an idea of how gem plugins work, check out the `sample_plugin` [README](https://github.com/lolcommits/lolcommits-sample_plugin) / [RubyDocs](http://www.rubydoc.info/github/lolcommits/lolcommits-sample_plugin) or browse through [other plugin](https://github.com/lolcommits) code.

The main reasons for extraction were to:

* remove the ever-growing gem dependency list in the main lolcommits gem
* start to unify configuration code for plugins (common helpers, approaches to OAuth etc.)
* make lolcommit plugin development easier
* add more comprehensive plugin tests

That's it, any questions just let me know.

Happy coding!

Matt

**Pull request closed without merging**